frontend/src/SignInForm/SigninForm.tsx

```diff
@@ -39,12 +41,22 @@ const useStyles = makeStyles((theme: Theme) =>
   })
 );
 
-function SigninForm({ signInClick, cancelClick, open }: ISigninFormProps) {
+function SigninForm({
+  signInClick,
+  cancelClick,
+  open,
+  error,
+}: ISigninFormProps) {
   const [showPassword, setShowPassword] = useState<boolean>(false);
-  const [email, setEmail] = useState<string>("")
-  const [password, setPassword] = useState<string>("")
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [emailIsValid, setEmailIsValid] = useState<boolean>(false);
+  const [passwordIsValid, setPasswordIsValid] = useState<boolean>(false);
 
   const classes = useStyles();
+  const emailPattern = new RegExp(
+    /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
```

Comment:
I can't read regex but do two-letter TLDs like "sibo@angular.io" work?

do dots and dashes like "sibo.yang-backup@email.com" work?

Reply:
They both work
![Screen Shot 2021-01-27 at 7 30 04 PM](https://user-images.githubusercontent.com/59661782/105985194-1e25e580-60d6-11eb-8bf9-91da52da45e3.png)
![Screen Shot 2021-01-27 at 7 30 19 PM](https://user-images.githubusercontent.com/59661782/105985218-24b45d00-60d6-11eb-87ff-fd2562dbb28b.png)